shippers/es.py

```diff
@@ -142,3 +147,37 @@ def flush(self) -> Any:
             self._log_outcome(success=success, failed=failed)
 
         self._bulk_actions = []
+
+    def discover_dataset(self, event: Dict[str, Any]) -> None:
+        if self._dataset == "":
+            body: str = event["Records"][0]["body"]
+            json_body: Dict[str, Any] = json.loads(body)
+            s3_object_key: str = ""
+
+            if "Records" in json_body and len(json_body["Records"]) > 0:
+                if "s3" in json_body["Records"][0]:
+                    s3_object_key = json_body["Records"][0]["s3"]["object"]["key"]
+                else:
+                    raise KeyError("Invalid event structure")
+            else:
+                raise KeyError("Invalid event structure")
+
+            if s3_object_key == "":
+                raise ValueError("S3 object key cannot be empty")
+
+            if (
+                "/CloudTrail/" in s3_object_key
+                or "/CloudTrail-Digest/" in s3_object_key
+                or "/CloudTrail-Insight/" in s3_object_key
+            ):
+                self._dataset = "aws.cloudtrail"
+            elif "vpcflowlogs" in s3_object_key:
+                self._dataset = "aws.vpc"
```

Comment:
this is the wrong dataset

you can find the correct one at https://github.com/elastic/integrations/blob/master/packages/aws/data_stream/vpcflow/sample_event.json#L5